### ISSUE TITLE
play.spotify.com: severe memory leaks scripts from `newrelic.com` are blocked

### ISSUE THREAD

**Issue by gorhill** [uBlockOrigin/uAssets#36]
### URL(s) where the issue occurs

`https://play.spotify.com/` (login required)

### Describe the issue

Original reporter:

> https://play.spotify.com/
> 
> more info https://community.spotify.com/t5/Help-D ... -p/1332839
> 
> it's occurs on adblock plus too

### Screenshot(s)

![a](https://cloud.githubusercontent.com/assets/585534/14853315/06dbd1b8-0c59-11e6-9500-9806cb9fff44.png)

### Versions

- Browser/version: Chome 51
- uBlock Origin version: 1.6.9

### Settings

Default settings.

### Notes

Ideally, a solution which keep the blocking if `newrelic.com` scripts would be optimal, but this would require further investigative work.

For now, these exception filters work:

    @@||newrelic.com^$third-party,domain=play.spotify.com
    @@||nr-data.net^$third-party,domain=play.spotify.com

Not sure about the second one, but given the seriousness of the issue, I will add it and investigate further when I have time (which is definitely not now).

**Comment by IDKwhattoputhere** [comment]
Related?
https://hg.adblockplus.org/easylist/rev/7c26684f1bee

**Comment by gorhill** [comment]
Is the reason for the change documented somewhere?

**Comment by IDKwhattoputhere** [comment]
Can't find it. Just noticed the change while browsing hg.adblockplus.org.

**Comment by gorhill** [comment]
> Edit: https://forums.lanik.us/viewtopic.php?p=92447#p92447

Topic does not exist.

Anyway, I disabled _uBlock filters -- unbreak_, added `||imasdk.googleapis.com/js/core/bridge*.html$subdocument,important,domain=spotify.com`, and the leak is still there.

I wish I had more time to dig into this right now, but only thing I got so far is:

![a](https://cloud.githubusercontent.com/assets/585534/14856441/0b7ed9b0-0c66-11e6-93ed-ff370819bbe7.png)

So it's newrelic.com itself which is the issue, but I am more inclined to think that the site is misusing newrelic.com's api, though at this point it's such a serious issue that un-blocking newrelic.com is the best quick fix until the real issue is pin-pointed in the site's code.

**Comment by gorhill** [comment]
After investigating more, as far as I can tell, newrelic.com's code is the culprit, not spotify.com. newrelic installs wrapper code around many native js calls (addEventListener, setTimeout, XMLHttpRequest.prototype.send, etc.) using inline script code on the main page and turns out it appears one of their wrapper's purpose is to log stats in a buffer (reason: [corporate bs placeholder]).

Presumably this buffer is processed/emptied by code in javascript code loaded separately. But the newrelic.com javascript files are blocked by default when using EasyPrivacy. This means that buffer is never emptied and grow indefinitely. End result: computer end up paging memory to disk, OS seize, etc.)

This is why the internet is so often crappy these days, because of such moronic 3rd parties, and also because 1st parties will eagerly believe the pitch of such 3rd parties to the point of blindly and recklessly importing their code in their own pages.

Correct me someone if I concluded wrongly.
